test/e2e/shared/aws.go

```diff
@@ -410,19 +410,24 @@ func createCloudFormationStack(prov client.ConfigProvider, t *cfn_bootstrap.Temp
 	cfnSvc := cloudformation.NewService(CFN)
 
 	Eventually(func() bool {
-		err := cfnSvc.ReconcileBootstrapStack(t.Spec.StackName, *renderCustomCloudFormation(t), tags)
+		err := cfnSvc.ReconcileBootstrapStack(t.Spec.StackName, *renderCustomCloudFormation(t), tags, true)
 		output, err1 := CFN.DescribeStackEvents(&cfn.DescribeStackEventsInput{StackName: aws.String(t.Spec.StackName), NextToken: aws.String("1")})
+		By("========= Stack Event Output Begin =========")
 		for _, event := range output.StackEvents {
 			By(fmt.Sprintf("Event details for %s : Resource: %s, Status: %s, Reason: %s", aws.StringValue(event.LogicalResourceId), aws.StringValue(event.ResourceType), aws.StringValue(event.ResourceStatus), aws.StringValue(event.ResourceStatusReason)))
 		}
+		By("========= Stack Event Output End =========")
 		return err == nil && err1 == nil
 	}, 2*time.Minute).Should(Equal(true))
+
 	stack, err := CFN.DescribeStacks(&cfn.DescribeStacksInput{StackName: aws.String(t.Spec.StackName)})
 	if err == nil && len(stack.Stacks) > 0 {
 		deleteMultitenancyRoles(prov)
 		if aws.StringValue(stack.Stacks[0].StackStatus) == cfn.StackStatusRollbackFailed ||
 			aws.StringValue(stack.Stacks[0].StackStatus) == cfn.StackStatusRollbackComplete ||
-			aws.StringValue(stack.Stacks[0].StackStatus) == cfn.StackStatusRollbackInProgress {
+			aws.StringValue(stack.Stacks[0].StackStatus) == cfn.StackStatusRollbackInProgress ||
+			aws.StringValue(stack.Stacks[0].StackStatus) == cfn.StackStatusCreateFailed ||
+			aws.StringValue(stack.Stacks[0].StackStatus) == cfn.StackStatusDeleteFailed {
```

Comment:
If the delete failed, we try deleting again? Do we expect it work the second time?

Reply:
Maybe yes. This is aws... but I can remove these. 😁

Reply:
Let's revisit after merging.